### PR TITLE
Implement configurable working directory

### PR DESCRIPTION
**Pull request by corvus-ch**
Implementation for #214

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update tests.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

### CODE REVIEW

**Comment by corvus-ch** [comment]
I guess we hit https://github.com/PyCQA/pylint/issues/214. @simu Do you have a creative idea to mitigate?

**Comment by simu** [comment]
> I guess we hit [PyCQA/pylint#214](https://github.com/PyCQA/pylint/issues/214). @simu Do you have a creative idea to mitigate?

The suggestion about configuring `min-similarity-lines` in the linked issue seems to work fine. To test it locally, I've added

```
[SIMILARITIES]
min-similarity-lines=10
```

to the `[pylint]` section in the `tox.ini` file.

File: commodore/cli.py [hunk 1]
```diff
@@ -30,9 +32,17 @@ def _version():
 @click.group()
 @click.version_option(_version(), prog_name="commodore")
 @verbosity
+@click.option(
+    "-d",
+    "--working-dir",
+    default="./",
+    show_default=True,
+    type=click.Path(file_okay=False, dir_okay=True),
+    help="Working directory where all the output will be written to.",
```
Comment (simu) [inline commodore/cli.py:41]:
```suggestion
    help="The directory in which Commodore will fetch dependencies, inventory and catalog, and store intermediate outputs",
```

File: docs/modules/ROOT/pages/reference/cli.adoc [hunk 2]
```diff
@@ -5,6 +5,11 @@
 Some command line options aren't associated with a specific command, but can
 provided to any command instead.
 
+*--working-dir*::
+  Commodore will create a whole bunch of directories and files.
```
Comment (simu) [inline docs/modules/ROOT/pages/reference/cli.adoc:9]:
```suggestion
  Commodore creates a large amount of directories and files.
```

File: docs/modules/ROOT/pages/reference/cli.adoc [hunk 3]
```diff
@@ -5,6 +5,11 @@
...
+  By default, this happens within the current working directory.
```
Comment (simu) [inline docs/modules/ROOT/pages/reference/cli.adoc:10]:
```suggestion
  By default, this happens in the current working directory of the environment in which Commodore is executed.
```

File: docs/modules/ROOT/pages/reference/cli.adoc [hunk 4]
```diff
@@ -5,6 +5,11 @@
...
+  Setting option, makes commodore create them at the provided location.
```
Comment (simu) [inline docs/modules/ROOT/pages/reference/cli.adoc:11]:
```suggestion
  If this option is provided, Commodore will create its directories and files in the provided location.
```

**Review by simu — APPROVED** [review]
LGTM, some changes in wording for the documentation of the new command line option.

**Comment by corvus-ch** [comment]
This PR implies a new rule to coding: never ever change working directories! Can we enforce that rule?

**Review by simu — APPROVED** [review]
LGTM now, I've ran a quick set of manual checks with `commodore -d tmp {catalog compile, component new, component delete, catalog compile --local}`, and everything seems to work as expected.

**Comment by simu** [comment]
> This PR implies a new rule to coding: never ever change working directories! Can we enforce that rule?

I spent 10 minutes googling for "python lint prohibit function use", but didn't find anything helpful yet.

**Review by srueg — APPROVED** [review]

**Pull request merged**